File: CPP/coolakt/392. Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int f=0;
        int l=0;
        int z=s.size();
        while(f<=s.size() && l<=t.size()){
            if(s[f]==t[l]){
                f++;
                l++;
                z--;
            }
            else{
                l++;
            }
        }
        if(z<=0){
            return true;
        }
        return false;
    } 
};

File: Java/Akash-212/392. Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
     int sIndex = 0; 

        for (int i = 0; i < t.length() && sIndex < s.length(); i++) {
            if (s.charAt(sIndex) == t.charAt(i)) {
                sIndex++; 
            }
        }

        return sIndex == s.length();
 
    }
}
